scripts/publish.ts

```diff
@@ -0,0 +1,104 @@
+  const defaultBranchName = options?.defaultBranchName ?? "main";
+
+  const args = mri<{ "dry-run": boolean }>(process.argv.slice(2), {
+    alias: { n: "dry-run" },
+    boolean: ["dry-run"],
+    default: { "dry-run": false },
+  });
+
+  const mode = args._[0];
+  if (mode !== "stable" && mode !== "unstable") {
+    console.error("The first argument must be either `stable` or `unstable`.");
+    process.exit(1);
+  }
+  const specifiers = args._.slice(1);
+  if (mode === "stable" && specifiers.length === 0) {
+    console.error(
+      "You must specify at least one package version (using the `<name>@<version>` format) to create a stable release.",
+    );
+    process.exit(1);
+  } else if (mode === "unstable" && specifiers.length > 0) {
+    console.error(
+      "You can't specify any package version when creating an unstable release.",
+    );
+    process.exit(1);
+  }
+
+  if (mode === "stable") {
+    const workspaceNames = await getWorkspaceNames();
+    const packageNameToSpecifiedVersion = new Map(
+      specifiers.map((specifier) => {
+        const parsedSpecifier = parsePackageNameSpecifier(specifier);
+        if (parsedSpecifier === undefined) {
+          console.error(
+            `Invalid package specifier: \`${specifier}\`. You must use the \`<name>@<version>\` format.`,
+          );
+          process.exit(1);
+        }
+        const [packageName, packageVersion] = parsedSpecifier;
+        if (!workspaceNames.has(packageName)) {
+          console.error(
+            `Invalid package specifier: \`${specifier}\`. \`${packageName}\` must be a non-private workspace.`,
+          );
+          process.exit(1);
+        }
+        const validPackageVersion = semverValid(packageVersion);
+        if (validPackageVersion === null) {
+          console.error(
+            `Invalid package specifier: \`${specifier}\`. Version \`${packageVersion}\` must be a valid SemVer string.`,
+          );
+          process.exit(1);
+        } else if (semverPrerelease(validPackageVersion) !== null) {
+          console.error(
+            `Invalid package specifier: \`${specifier}\`. Version \`${validPackageVersion}\` mustn't include any prerelease component.`,
+          );
+          process.exit(1);
+        }
+        return [packageName, validPackageVersion];
+      }),
+    );
+    console.log(packageNameToSpecifiedVersion); // TODO: remove this line.
+  }
+
+  const { stdout: status } = await $`git status --porcelain`;
+  if (status.length > 0) {
+    console.error("This command must be executed on a clean repository.");
+    process.exit(1);
+  } else if (mode === "stable") {
+    const { stdout: currentBranchName } = await $`git branch --show-current`;
+    if (currentBranchName !== defaultBranchName) {
+      console.error(
+        `You must be on the \`${defaultBranchName}\` branch to create a stable release.`,
+      );
+      process.exit(1);
+    }
+  }
+}
+
+async function getWorkspaceNames(): Promise<Set<string>> {
```

Comment:
Any ideas on how this script could be made package manager agnostic? The script currently relies on Yarn and workspaces, but this release workflow would be useful in our other packages, like `@prismicio/client`, that don't use those.

We don't need and answer now or support that any time soon, but it could be good to consider. 🙂 I'm pretty sure you already mentioned this idea.